content/en/docs/concepts/overview/object-management-kubectl/kustomization.md

```diff
@@ -0,0 +1,752 @@
+```yaml
+apiVersion: v1
+data:
+  password: c2VjZXJ0
+  username: YWRtaW4=
+kind: Secret
+metadata:
+  name: example-secret-2-t52t6g96d8
+type: Opaque
+```
+
+#### generatorOptions
+The generated ConfigMaps and Secrets hvae a suffix appended by hashing the contents. This ensures that a new ConfigMap or Secret is generated when the content is changed. To disable the behavior of appending a suffix, one can use `generatorOptions`. Besides that, it is also possible to specify cross-cutting options for generated ConfigMaps and Secrets.
+```shell
+cat <<EOF >./kustomization.yaml
+configMapGenerator:
+- name: example-configmap-3
+  literals:
+  - FOO=Bar
+generatorOptions:
+  disableNameSuffixHash: true
+  labels:
+    type: generated
+  annotations:
+    note: generated
+EOF
+```
+By `kubectl kustomize ./`, you can see the generated ConfigMap as
+```yaml
+apiVersion: v1
+data:
+  FOO: Bar
+kind: ConfigMap
+metadata:
+  annotations:
+    note: generated
+  labels:
+    type: generated
+  name: example-configmap-3
+```
+
+### Setting cross-cutting fields
+It is quite common to set some cross-cutting fields at the same time for all
+Kubernetes Resources that are in the same project
+* setting the same namespace for all Resource
+* adding the same name prefix or suffix
+* adding the same set of labels
+* adding the same set of annotations
+
+Here is an example:
+```shell
+# Create a deployment.yaml
+cat <<EOF >./deployment.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+spec:
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx
+EOF
+
+cat <<EOF >./kustomization.yaml
+namespace: my-namespace
+namePrefix: dev-
+nameSuffix: "-001"
+commonLabels:
+  app: bingo
+commonAnnotations:
+  oncallPager: 800-555-1212
+resources:
+- deployment.yaml
+EOF
+```
+By `kubectl kustomize ./`, you can see those fields are all set in the Deployment Resource.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    oncallPager: 800-555-1212
+  labels:
+    app: bingo
+  name: dev-nginx-deployment-001
+  namespace: my-namespace
+spec:
+  selector:
+    matchLabels:
+      app: bingo
+  template:
+    metadata:
+      annotations:
+        oncallPager: 800-555-1212
+      labels:
+        app: bingo
+    spec:
+      containers:
+      - image: nginx
+        name: nginx
+```
+
+### Composing and Customizing Resources
+It is common to compose a set of Resources in a project and management them inside
+the same file or directory. 
+Kustomize offers composing Resources from different files and applying patches or other customization to them. 
+
+#### Composing
+To compose different Resource files in `kustomization.yaml`, one need to add those files to the `resources` list.
+Here is an example for an nginx application with a Deployment and a Service.
```

Comment:
Does this help clarify instead of/addition line 240: 
Kustomize supports composition of different resources. The `resources` field, in the kustomization.yaml file, defines the list of resources to include in a configuration. Set the path to a resource's configuration file in the `resources` list.

Reply:
updated